[Documentation] Mention dialogs in Buttons doc (#10039)
marmelab/react-admin: docs/Buttons.md

**Tip**: If you want to link to the Edit view manually, use the `/{resource}/{record.id}` location.

> **Tip:** You are also able to use a [`<EditInDialogButton>`](./EditInDialogButton.md) to opening an `<Edit>` view inside a dialog, hence allowing to edit a record without leaving the current view.

Reviewer: ```suggestion
**Tip:** To allow users to edit a record without leaving the current view, use the [`<EditInDialogButton>`](./EditInDialogButton.md) component.
```

**Tip**: If you want to link to the Create view manually, use the `/{resource}/create` location.

> **Tip:** You are also able to use a [`<CreateInDialogButton>`](./CreateInDialogButton.md) to open a `<Create>` view inside a dialog, hence allowing to create a new record without leaving the current view.

Reviewer: ```suggestion
**Tip:** To allow users to edit a record without leaving the current view, use the [`<EditInDialogButton>`](./EditInDialogButton.md) component.
```
